torch/ao/quantization/fx/prepare.py

```diff
@@ -241,18 +243,14 @@ def is_pattern_dtype_config_supported_by_backend(
     for dtype_config in dtype_configs:
         # check if arg dtype are supported
         supported = True
-        for arg in input_node.args:
-            supported = supported and \
-                is_input_arg_dtype_supported_by_backend(
-                    arg, input_node, node_name_to_target_dtype, dtype_config, backend_config)
-        for k, arg in input_node.kwargs.items():
-            supported = supported and \
-                is_input_arg_dtype_supported_by_backend(
-                    arg, input_node, node_name_to_target_dtype, dtype_config, backend_config)
+        for arg in list(input_node.args) + list(input_node.kwargs.values()):
+            supported = supported and is_input_arg_dtype_supported_by_backend(
```

Comment:
I feel ideally the qmin/qmax scale_min/scale_max range check should happen in `is_input_arg_dtype_supported_by_backend` function since it's part of the dtype

Reply:
Ok, that would require us to pass the QConfig in there. Is that OK?

Reply:
Actually, I'm not sure, since we'd be validating against the same `dtype_config` multiple times unnecessarily

Reply:
that's what we do in these functions already, since `dtype_config` is a config for all input args and all output, it's OK to validate different parts separately I think

Reply:
can we pass in `qconfig.activation` or `qconfig.activation()` here? similar for other callsites

Reply:
basically think about how are you going to extend this if we have arbitrary args and kwargs and outputs, it would make more sense to have a single function to validate these things for each arg

Reply:
Ok, I pushed the activation validation logic into `is_input_arg_dtype_supported_by_backend` and `is_output_dtype_supported_by_backend`. I left the weight validation logic here since the above two functions only deal with activations.